packages/react-dom/src/__tests__/ReactDOMFiber-test.js

```diff
@@ -1272,4 +1283,48 @@ describe('ReactDOMFiber', () => {
     });
     expect(didCallOnChange).toBe(true);
   });
+
+  it('should restore selection in the correct window', async () => {
+    // creating new JSOM instance to get a second window as window.open is not implemented
```

Comment:
nit: JSDOM

Reply:
Done in https://github.com/facebook/react/pull/30951/commits/9e79394f2d7c75fe9458f2dcb425bba812bd4a04